File: biddo_client/lib/core/controllers/review.dart
import 'dart:async';

import 'package:get/get.dart';
import '../models/account.dart';
import '../models/review.dart';
import '../repositories/review.dart';
import 'account.dart';

class ReviewsController extends GetxController {
  final reviewRepository = Get.find<ReviewRepository>();
  final accountController = Get.find<AccountController>();

  RxList<Account> accountsWithMostReviews = <Account>[].obs;
  RxBool accountsWithMostReviewsLoading = false.obs;

  Future<String?> translate(String id, String lang) async {
    return await reviewRepository.translate(id, lang);
  }

  Future<List<Review>> loadForAccount(String accountId, int page, int perPage) {
    return reviewRepository.loadForAccount(accountId, page, perPage);
  }

  Future<Review?> saveReview(
    double stars,
    String auctionId, [
    String? description,
    String? id,
  ]) {
    return reviewRepository.saveReview(
      stars,
      auctionId,
      description,
      id,
    );
  }
}

File: core/controllers/review.ts
import { ReviewRepository } from '../repositories/review'

class ReviewsController {
  public async save(params: {
    reviewId?: string
    auctionId: string
    stars: number
    description?: string
  }) {
    try {
      return await ReviewRepository.saveReview(params)
    } catch (error) {
      console.error(`Could not save review: ${error}`)
      return null
    }
  }

  public async translate(reviewId: string, language: string) {
    try {
      return await ReviewRepository.translateReview(reviewId, language)
    } catch (error) {
      console.error(`Could not translate review: ${error}`)
      return null
    }
  }
  public async getForAccount(accountId: string, page: number, perPage: number) {
    try {
      return await ReviewRepository.loadForAccount(accountId, page, perPage)
    } catch (error) {
      console.error(`Could not load reviews for account: ${error}`)
      return []
    }
  }

  public async getForLoggedAccount(page: number, perPage: number) {
    try {
      return await ReviewRepository.loadForLoggedAccount(page, perPage)
    } catch (error) {
      console.error(`Could not load reviews for logged account: ${error}`)
      return []
    }
  }
}

const revewsController = new ReviewsController()
export { revewsController as ReviewsController }
